src/components/pages/main/components/bookDetails/BookDetails.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useRef } from "react";
+import "./BookDetails.css";
+import { useNavigate, useParams } from "react-router-dom";
+import { IBookDetails } from "../../../../../interfaces";
+import { apiService } from "../../../../../services/ApiService";
+import { FetchStatusEnum } from "../../../../../enums/FetchStatusEnum";
+import useOutsideAlerter from "../../../../../utils/useOutsideAlerter";
+
+interface BookDetailsProps {}
+
+const BookDetails: React.FC<BookDetailsProps> = () => {
+  const params = useParams<{ bookId: string }>();
+
+  const [bookDetails, setBookDetails] = React.useState<IBookDetails>(
+    {} as IBookDetails,
+  );
+  const [fetchStatus, setFetchStatus] = React.useState<FetchStatusEnum>(
+    FetchStatusEnum.loading,
+  );
+
+  const navigate = useNavigate();
+  const wrapperRef = useRef(null);
+  useOutsideAlerter(wrapperRef, () => navigate(-1));
+
+  React.useEffect(() => {
+    const getBookDetails = async () => {
+      try {
+        const data = await apiService.getBookById(
+          params?.bookId?.toString() || "",
+        );
+        setBookDetails(data as IBookDetails);
+        setFetchStatus(FetchStatusEnum.done);
+      } catch (error) {
+        setFetchStatus(FetchStatusEnum.error);
+        throw error;
+      }
+    };
+    getBookDetails();
+  }, [params?.bookId]);
+
+  if (fetchStatus === "loading") {
+    return (
+      <div className="loader-wrapper">
+        <div className="loader"></div>
+      </div>
+    );
+  }
+  if (fetchStatus === "error") {
+    return <div>ERR!</div>;
+  }
+
+  return (
+    <>
+      {fetchStatus === "done" && Object.keys(bookDetails).length !== 0 && (
```

Comment:
```suggestion
      {fetchStatus === FetchStatusEnum.done && Object.keys(bookDetails).length  && (
```
старайся избегать сравнений типа !== 0 или === "". используй их как фолси значения